Update website structure to make navigation between sessions easier

Paulj1989 opened the pull request.

Paulj1989 commented:
@ed-chick @jonaswillemsen @EmileJH 

I had a first run at reworking the site to better cater to people navigating session materials. It's definitely not finished, but it would be good to get some feedback.

ed-chick commented:
I'd prefer the text on the home page, Glossary page and Schedule page to be left-aligned. This would make it consistent with the Sessions materials (and it's a personal preference). (I also find the contrast between the narrow text in the middle of the Schedule and Glossary pages & the tables little jarring).

The Resources section no longer seems to be accessible. Is this just because the NCF page was the only one we had so far & this has been moved in with the Schedule?

I'm also not sure about the home page in the new format. The old format filled the screen better. There's a lot of unused space and the need to scroll (I suppose it depends on which screen you are using).

Last thing & I'm aware we are trying to cover two different aspects here, but are we certain that the Sessions menu structure is how we want it? It's going in order of session, but it also has the module as the top layer, currently meaning that Visualisation is repeated. I think that maybe we drop the session number and just put them under the module names. In the future, our sessions schedule is likely to be more flexible when we start bringing in guest speakers & we might not want a formal session page for all of them.

ed-chick commented:
I do generally agree with the structural change, though i.e. navigation to the main areas of the site along the top & then sub-pages down the left.

Paulj1989 commented:
Thanks for the feedback @ed-chick! I've grouped the feedback by broad topic and responded below.

## Formatting

> I'd prefer the text on the home page, Glossary page and Schedule page to be left-aligned. This would make it consistent with the Sessions materials (and it's a personal preference). (I also find the contrast between the narrow text in the middle of the Schedule and Glossary pages & the tables little jarring).

Consistency in style at each layer in the hierarchy makes sense. I'm unsure how well left-alignment would work given the tables, especially the Schedule table (because it is so wide). One solution would be to drop some of the wider columns in the table so that the table fits the page better. The GitHub Link page can probably go (I think it makes sense to add the Nectar Relaunch slides to this repo), and maybe the recordings should be available via the page for each session? 

Another option would be to pull these pages into the sessions section (the glossary could go in an appendix, and the schedule could be the landing page when you click on the Sessions link, and could be available in the sidebar at the very beginning.

I'm not sure which of these would be better. The glossary probably belongs in an appendix, but I could see an argument for either approach with the Schedule page. Either way, I broadly agree, and I'm happy to have a play around to see what works best (but welcome your view on this question).

> I'm also not sure about the home page in the new format. The old format filled the screen better. There's a lot of unused space and the need to scroll (I suppose it depends on which screen you are using).

I'd push back a little on this. The homepage should be as simple as possible. Generally speaking (and especially with a static site like this), the homepage should tell a visitor exactly what the site is doing and nothing more, and navigation to pages with all other relevant information should be straightforward. Filing the screen is probably a sign that this goal isn't being achieved.

However, I think the use of space could be better (and in terms of use of space, the original layout was better). It's possible you are referring to this, and I've misunderstood. I think I need to use more width so visitors won't need to scroll to see the whole page (on the average laptop screen), but part of that is potentially reducing the amount on the homepage, too. Either that means some of the information should go on an About page, or it means editing the current text and reducing it a little.

The logo is my biggest issue with the homepage (besides the fact that it requires scrolling). I liked the colour of the sidebar and think the logo sat quite nicely in the sidebar, but the logo I've had to use with this setup is a little less appealing. Unfortunately, I don't think there is an obvious solution. Perhaps a redesign of the logo (or some new colours)?

## Site Structure 

> The Resources section no longer seems to be accessible. Is this just because the NCF page was the only one we had so far & this has been moved in with the Schedule?

Yeah, I added the NCF competencies to the Schedule because that seemed to make more sense than having a second schedule that included them. I've removed the resources section for now, just because there was nothing else in it, but if there are additional resources that we want to add to the site, that's fine, and the resources directory can be added back in.

> Last thing & I'm aware we are trying to cover two different aspects here, but are we certain that the Sessions menu structure is how we want it? It's going in order of session, but it also has the module as the top layer, currently meaning that Visualisation is repeated. I think that maybe we drop the session number and just put them under the module names. In the future, our sessions schedule is likely to be more flexible when we start bringing in guest speakers & we might not want a formal session page for all of them.

I think following the schedule order would be most intuitive for users. It would also avoid any confusion regarding navigation because active pages would all be grouped without any gaps. This would make it intuitively clear where we are in the course. Perhaps we could change the module name for the Streamlit Dashboards session? Something like Front-End Development? Or it could just be Visualisations (Cont.) or something like that?

Paulj1989 committed:
Adjust page layout and table formatting

Paulj1989 committed:
Use full page layout

Paulj1989 commented:
I've made several changes to reflect some of the feedback:

- The homepage still needs work, but the use of space is a little better
- The schedule and glossary pages changes reflect the formatting concerns, and I think it is an improvement. However, there's still a question about what information is included in both and where they should sit exactly.

jonaswillemsen commented:
Overall I think the new structure looks great - I will say that I agree we should have some sort of landing page when the user opens the Sessions page, rather than it going straight into the page for session 1.

I think a Resources section does still make sense, and I wonder if the Glossary page could go in there rather than being its own top-level page?

Paulj1989 commented:
> Overall I think the new structure looks great - I will say that I agree we should have some sort of landing page when the user opens the Sessions page, rather than it going straight into the page for session 1.

Yeah, I felt the same when I set it up. I am reluctant to create a landing page for the sake of it (or any page for the sake of it), but if the schedule page makes sense as the landing page that works for me. The only question is whether it would complicate the user's experience more than necessary.

> I think a Resources section does still make sense, and I wonder if the Glossary page could go in there rather than being its own top-level page?

I'd have no issue with this. The resources section only got dropped because there was nothing in it right now, but I'm happy enough to chuck the Gloassry page in there.

Paulj1989 committed:
Remove NCF page

Paulj1989 committed:
Move Glossary page to Resources section

Paulj1989 committed:
Update Glossary page formatting

Paulj1989 commented:
- I've moved the glossary page to the resources section. The resources section is currently a dropdown from the navbar, which I don't love aesthetically (the dropdown itself looks a little clunky), but I'm not sure there's a better way to structure that section.
    - The answer is probably cleaning up the dropdown in the CSS, but not entirely sure what it is that I don't like about it, so will have a think.
- I've also shifted links to the 'Term' column in the glossary. I figured this made more sense and cleared space for the definitions.

ed-chick commented:
Hi Paul,

Thanks very much for all of this. A few further thoughts:

We have lost the Recording Link from the Schedule but the text still mentions it.

Should we still have a Welcome! link in the top-level navigation bar? I realise the logo in the top-left serves this purpose, but feel that it might help with site navigation. People might want to navigate back to the home page
when they send the link to colleagues.

I'm happy with the site structure.

I think the centrally-aligned text looks better now that the column is a bit wider. 

Yes, when I said that the hope page previously "filled the screen" better, I meant in terms of proportions/layout, rather than overfilling it with features and text.

Regarding removing GitHub repo links, are we sure we want to do this? Future sessions will have Jupyter Notebooks for people to work through and play with and people will need to be directed to where they can obtain them. Unless we can embed interactive Jupyter Notebooks into the webpage. We could have the links on each Session page, rather than in the Schedule.

On reflection, we can probably leave the Sessions side-bar as is, keeping them in numeric order since it is more intuitive. Repeating "Visualisation" is no big deal.

Paulj1989 committed:
Move Schedule to Sessions section

Paulj1989 committed:
Increase navbar logo size

Paulj1989 commented:
> We have lost the Recording Link from the Schedule but the text still mentions it.

I've removed this section, but maybe it needs a little detail explaining that each session's materials are available on its dedicated page under the Sessions section. 
 
> Should we still have a Welcome! link in the top-level navigation bar? I realise the logo in the top-left serves this purpose, but feel that it might help with site navigation. People might want to navigate back to the home page when they send the link to colleagues.

I've added a Home page to the navbar. I assume most people will know to click the logo to navigate back to the homepage, but this should make sure. I have also increased the navbar logo size, just because it was a little too small already.

I have also moved the Schedule page to the landing page for the Sessions section. I'm still unsure whether this is better (I prefer it aesthetically, but don't want to make it too hard for people to find).
 
> Regarding removing GitHub repo links, are we sure we want to do this? Future sessions will have Jupyter Notebooks for people to work through and play with and people will need to be directed to where they can obtain them. Unless we can embed interactive Jupyter Notebooks into the webpage. We could have the links on each Session page, rather than in the Schedule.

I think any materials for Code Club should probably be a part of the repository, so a GitHub repository link wouldn't make a lot of sense. We can make notebooks available for download by [including other formats](https://quarto.org/docs/output-formats/html-multi-format.html#specifying-formats-to-link) in the page YAML. Another option is [using Binder](https://quarto.org/docs/projects/binder.html, but my concern is that it doesn't get people setting up their projects and getting familiar with the workflow.

Paulj1989 committed:
Clean up Schedule YAML

Paulj1989 committed:
Fix missing toc

Paulj1989 committed:
Reduce homepage text

Paulj1989 commented:
I've edited the text on the homepage to shorten it and ensure that everything fits without scrolling. @ed-chick, do you want to read it and let me know what you think? I think it retains the spirit of what was initially written.

I'm okay with adding an About page that includes more detail (including a nod to the history of Code Club, for example), but think the homepage does need to "fit to page".

jonaswillemsen committed:
fix markdown links in glossary

ed-chick commented:
@Paulj1989 , thanks for your work on this, I'm happy with it all & agree with the design choices.

@jonaswillemsen , thanks for updating the glossary.

jonaswillemsen committed:
quarto render to fix glossary

Paulj1989 committed:
Merge branch 'main' into feat/spring-cleaning-and-setup

Paulj1989 committed:
Use full page layout

EmileJH committed:
Update index.qmd

EmileJH committed:
Update slides.qmd

EmileJH committed:
updating

Paulj1989 commented on sessions/02-jupyter_notebooks/slides.qmd:
```diff
@@ -19,17 +19,14 @@ format: revealjs
 ## Pros and cons of using a notebook {.center}
 
-thyere are some notable pros and cons to using a Jupyter notebook, it is implortant to be aware of these before selecting them as the right tool for your task.
+There are some notable pros and cons to using a Jupyter notebook.
```
This slide doesn't quite fit on the page. It is overlapping with the footer. Might need to trim it slightly.

Paulj1989 replied:
If you are struggling to trim the content, removing the "There are some notable pros..." bit could probably go, since that can be inferred from the slide title.

EmileJH replied:
I've edited the slice

EmileJH committed:
Update to P&Q slide

The pros and cons slide has been updated to make points more concise and remove unnecessary text.

jonaswillemsen reviewed (approved):
looks good to me
